### PR TITLE
Implement adding config hash annotation to deployments

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does**:

Implement `annotateConfigHash` to use Kubernetes sync

**Why we need it**:

We add this annotation to ensure that the workload should be restarted when one of its configurations changes.

**Which issue(s) this PR fixes**:

Part of #4980 

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/app/pipedv1/plugin/kubernetes/deployment/annotate.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import (
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/provider"
+)
+
+// annotateConfigHash appends a hash annotation into the workload manifests.
+// The hash value is calculated by hashing the content of all configmaps/secrets
+// that are referenced by the workload.
+// This appending ensures that the workload should be restarted when
+// one of its configurations changed.
+func annotateConfigHash(manifests []provider.Manifest) error {
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/annotate.go:26]: I almost copied from below.
https://github.com/pipe-cd/pipecd/blob/e7cbc645c8028d016b40f5f08f559f7343092eb3/pkg/app/piped/executor/kubernetes/kubernetes.go#L587

File: pkg/app/pipedv1/plugin/kubernetes/deployment/annotate_test.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/provider"
+)
+
+func TestAnnotateConfigHash(t *testing.T) {
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/annotate_test.go:26]: I almost copied from below and added a testcase for Secrets
https://github.com/pipe-cd/pipecd/blob/e7cbc645c8028d016b40f5f08f559f7343092eb3/pkg/app/piped/executor/kubernetes/kubernetes_test.go#L655

File: pkg/app/pipedv1/plugin/kubernetes/provider/resource.go [hunk 3]
```diff
@@ -90,6 +90,16 @@ func MakeResourceKey(obj *unstructured.Unstructured) ResourceKey {
 	return k
 }
 
+func (k ResourceKey) IsDeployment() bool {
```
Comment (khanhtc1202) [inline pkg/app/pipedv1/plugin/kubernetes/provider/resource.go:93]: Could you add test for this 👀

**Review by khanhtc1202 — APPROVED** [review]
Let's go 🚀

**Review by ffjlabo — APPROVED** [review]
👍

**Pull request merged**